src/models/user.model.ts

```diff
@@ -0,0 +1,32 @@
+import { Schema, Document, model, PassportLocalSchema } from 'mongoose';
+import passportLocalMongoose from 'passport-local-mongoose';
+
+export interface IUser extends Document {
+	username: string;
+	email: string;
+	name: string;
+	isAdmin: boolean;
+}
+
+const UserSchema: Schema = new Schema({
+	name: {
```

Comment:
Keeping in mind the ways in which different pages in our app will use the information, wouldn't it be better to separate name to first name and last name? If our pages display the user's name, something simple such as 'Welcome Blah', it would be better to address always using the first name. How about a required first name and optional last name?

Reply:
Makes sense, I will change it to have first name and last name instead

Reply:
I think it's fine as it is. We can use username for greetings.